src/pathfinding_igvc/include/AStar.h

```diff
@@ -33,9 +100,91 @@ class AStar {
      * @return points stacked in order, where the top contains the starting
      * GridPoint and the bottom contains the goal GridPoint
      */
-    std::stack<GridPoint> run(nav_msgs::OccupancyGrid occupancy_grid,
+    static std::stack<GridPoint> run(nav_msgs::OccupancyGrid occupancy_grid,
                               GridPoint start,
                               GridPoint goal);
+
+private:
+    /**
+     * Initializes the class with the given occupancy_grid and start and
+     * goal grid points
+     *
+     * The constructor is private because a path would only be found once
+     * for a given occupancy grid and start and goal points.
+     *
+     * @param occupancy_grid occupancy grid
+     * @param start
+     * @param goal
+     */
+    AStar(nav_msgs::OccupancyGrid occupancy_grid, GridPoint start,
+          GridPoint goal);
```

Comment:
I'm a little confused why this constructor is private?

Reply:
already commented below, but copy pasting here:

If we were to make the constructor public, that means that we can create an AStar object and reuse it for finding path multiple times. First of all, I don't see any use to create an AStar object and find the path more than once with the exact same occupancy grid and starting/goal point. Second of all, even if we were to reuse the AStar object, we would need to ensure that all the instance variables in this class are reinitialized. If we just make the constructor private, we don't need to worry about the AStar object being created and reused multiple times without all of its instance variables being reinitialized.

I actually had this issue with DBSCAN and the bug was discovered pretty late since it only happened after the first callback. So making it private would avoid any issue of this sort

Let me know what you think/if you have any better suggestions!

Reply:
OK, I follow. Let's just make sure to put a (condensed) version of your explanation in a comment in the code itself so that people in the future understand why we're doing this.